### PR TITLE
feat: allow passing http serverOptions from config

### PR DESCRIPTION
**Pull request by innerdvations**
### What does it do?

Allows user to set http server configuration options

### Why is it needed?

The specific issue is that Strapi can no longer accept any file uploads that take longer than the default Node 18+ http server requestTimeout, which is 5 minutes.

This allows users to configure that setting to be whatever value they wish.

In addition to that, users will now be able to configure any option available in http.ServerOptions

### How to test it?

Try setting some values in config/server.ts http.serverOptions and see that they are passed through to the http.createServer

### Related issue(s)/PR(s)

Fixes https://github.com/strapi/strapi/issues/18543

[Documentation PR](https://github.com/strapi/documentation/pull/1903)

### CODE REVIEW

**Review by derrickmehaffy — APPROVED** [review]
Love it

**Comment by innerdvations** [comment]
> Love it

I do too, I think it would provide a lot of solutions to edge cases, and if we like this, I will probably do the same for Koa, Apollo, and any other external modules we use. We already do it with Knex.

Ultimately however this is a philosophical question that I'll need to wait for a consensus on before we can merge.

**Review by Marc-Roig — DISMISSED** [review]

**Review by pwizla — APPROVED** [review]

**Pull request merged**
